### PR TITLE
Implement prefix for saved mapping file

### PR DESCRIPTION
**Pull request by MindaugasLaganeckas**
Address #1037

### CODE REVIEW

File: src/WireMock.Net/Serialization/MappingFileNameSanitizer.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+using WireMock.Settings;
+
+namespace WireMock.Serialization;
+
+/// <summary>
+/// Creates sanitized file names for mappings
+/// </summary>
+public class MappingFileNameSanitizer
+{
+    private readonly char _replaceChar = '_';
```
Comment (StefH) [inline src/WireMock.Net/Serialization/MappingFileNameSanitizer.cs:13]: private const char ReplaceChar = '_';

File: test/WireMock.Net.Tests/Serialization/MappingFileNameSanitizerTests.cs [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+using Moq;
+using WireMock.Serialization;
+using WireMock.Settings;
+using Xunit;
+
+namespace WireMock.Net.Tests.Serialization;
+
+public class MappingFileNameSanitizerTests
+{
+    private readonly string _mappingGuid = "ce216a13-e7d6-42d7-91ac-8ae709e2add1";
```
Comment (StefH) [inline test/WireMock.Net.Tests/Serialization/MappingFileNameSanitizerTests.cs:11]: both can be const

File: test/WireMock.Net.Tests/Serialization/MappingFileNameSanitizerTests.cs [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    private readonly string _mappingTitle =
```
Comment (StefH) [inline test/WireMock.Net.Tests/Serialization/MappingFileNameSanitizerTests.cs:12]: place this on 1 line

**Review by StefH — CHANGES_REQUESTED** [review]

**Comment by MindaugasLaganeckas** [comment]
Thank you for the feedback! I have addressed all comments and added an extra test for the usecase, that is going to be our primary (somehow missed in the initial implementation).

File: src/WireMock.Net/Serialization/MappingFileNameSanitizer.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    {
+        _settings = Guard.NotNull(settings);
+    }
+
+    /// <summary>
+    /// Creates sanitized file names for mappings
+    /// </summary>
+    public string BuildSanitizedFileName(IMapping mapping)
+    {
+        const string mappingTitlePrefix = "Proxy Mapping for ";
```
Comment (StefH) [inline src/WireMock.Net/Serialization/MappingFileNameSanitizer.cs:27]: Use the public const value in the ProxyAndRecordSettings class here.

File: src/WireMock.Net/Settings/ProxyAndRecordSettings.cs [hunk 5]
```diff
@@ -95,9 +95,14 @@ public string SaveMappingForStatusCodePattern
     /// </summary>
     public bool AppendGuidToSavedMappingFile { get; set; }
 
+    /// <summary>
+    /// Set prefix for saved mapping file.
+    /// </summary>
+    public string PrefixForSavedMappingFile { get; set; } = "Proxy Mapping for ";
```
Comment (StefH) [inline src/WireMock.Net/Settings/ProxyAndRecordSettings.cs:101]: Create a public const for this the value in this class.

File: test/WireMock.Net.Tests/Serialization/MappingFileNameSanitizerTests.cs [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+
+    [Fact]
+    public void BuildSanitizedFileName_WithTitleAndGuid_NoAppendGuidSetting()
+    {
+        // Arrange
+        var mappingMock = new Mock<IMapping>();
+        mappingMock.Setup(m => m.Title).Returns(MappingTitle);
+        mappingMock.Setup(m => m.Guid).Returns(new Guid(MappingGuid));
+
+        var settingsMock = new WireMockServerSettings
```
Comment (StefH) [inline test/WireMock.Net.Tests/Serialization/MappingFileNameSanitizerTests.cs:68]: officially, this is not mocked, so the name can be `settings`

**Review by StefH — CHANGES_REQUESTED** [review]

**Review by StefH — APPROVED** [review]

**Comment by MindaugasLaganeckas** [comment]
@StefH if it is not too much trouble, may I ask you to release the feature? I would like to start using it as soon as possible  😄

**Comment by StefH** [comment]
New NuGet will be released today or tomorrow.

**Comment by StefH** [comment]
NuGet is released

**Comment by MindaugasLaganeckas** [comment]
I will switch to the latest Wiremock.Net right away for my demo today. Thank you for expediting the release! 🎅

**Comment by StefH** [comment]
Ah cool. A demo for your company?

**Comment by MindaugasLaganeckas** [comment]
More like a demo for the team inside the company 😄

**Pull request merged**